core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -787,6 +787,33 @@ sealed abstract private[data] class NonEmptyListInstances extends NonEmptyListIn
       )(f: A => G[B])(implicit G: Applicative[G]): G[NonEmptyList[B]] =
         fa.traverse(f)
 
+      override def mapAccumulate[S, A, B](init: S, fa: NonEmptyList[A])(f: (S, A) => (S, B)): (S, NonEmptyList[B]) = {
+        var state = init
+
+        val fb = fa.map { a =>
+          val (newState, b) = f(state, a)
+          state = newState
+          b
+        }
+
+        (state, fb)
+      }
```

Comment:
I'm wondering if we could simply re-use implementations from regular collections for their non-empty counterparts. Something like
```scala
// Assume there's `listTraverse: Traverse[List]` accessible somehow:
override def mapAccumulate[S, A, B](init: S, fa: NonEmptyList[A])(f: (S, A) => (S, B)): (S, NonEmptyList[B]) = {
  val (last, fb) = listTraverse.mapAccumulate[A, A, B](init, fa.toList)(f)
  (last, NonEmptyList.fromListUnsafe(fb)) // should be safe, actually
}
```
And the same for all other NEs – would it make sense, wdyt?

Reply:
I thought about this but decided to go the dumb route first and see what you guys thought about it.
I think I prefer Oscar's idea of having a unified implementation that just requires `Functor`.

Reply:
Yes, agree. That one looks even better to be.